src/java.base/share/classes/java/io/File.java

```diff
@@ -1621,6 +1621,13 @@ public boolean setWritable(boolean writable) {
      * file attributes including file permissions. This may be used when finer
      * manipulation of file permissions is required.
      *
+     * <p> If setting the read permission is supported, but the user does not
+     * have permission to change the access permissions of this abstract
+     * pathname, then the operation will fail. If the platform or underlying
+     * file system does not support setting a file's read permission, this
+     * method does nothing and returns the value of the {@code readable}
+     * parameter.
+     *
```

Comment:
I think we are close to find the right wording for this. The return description looks fine. For the method description, I have two suggested changes:

1.  "If setting the read permission is supported" -> "If the platform supports settings a file's read permission".

2. "If the platform or underlying  file system does not support" -> "If the platform does not support". The reason for this is that the underlying file systems may or may support file permissions, an implementation can't distinguish those.